### ISSUE TITLE
DOT and DOH didn't work

### ISSUE THREAD

**Issue by SpirioSpirio** [0xERR0R/blocky#1298]
Hello,

I run a blocky in a podman pod on dns.spirio.fr with a certificate generate by let'sencrypt and certbot. It works properly on port 53 but it doesn't work on 853 and 443 whereas there is a certificate and blocky run.

These is the docker-compose configuration.

```
version: "2.1"
services:
  blocky:
    image: spx01/blocky
    container_name: blocky
    restart: unless-stopped
    # Optional the instance hostname for logging purpose
    hostname: blocky-hostname
    ports:
      - "54.37.234.80:53:53/tcp"
      - "54.37.234.80:53:53/udp"
      - "54.37.234.80:853:853/tcp"
      - "54.37.234.80:853:853/udp"
      - "127.0.0.1:8081:443/tcp"
    volumes:
      # Optional to synchronize the log timestamp with host
      - /etc/localtime:/etc/localtime:ro
      # config file
      - /etc/blocky/blocky.yml:/app/config.yml
      - /etc/blocky/whitelist.txt:/etc/blocky/whitelist.txt
      - /etc/blocky/blacklist.txt:/etc/blocky/blacklist.txt
      - /etc/letsencrypt/live/dns.spirio.fr/fullchain.pem:/etc/blocky/fullchain.pem
      - /etc/letsencrypt/live/dns.spirio.fr/privkey.pem:/etc/blocky/privkey.pem
    user: "1000:1000"
    dns:
      - 1.1.1.1
      - 9.9.9.9
```

This is my apache configuration :

```

<VirtualHost *:80>
        ServerName dns.spirio.fr
        ServerAdmin contact@spirio.fr

        ErrorLog /var/log/httpd/002-dns.spirio.fr/error.log
        CustomLog /var/log/httpd/002-dns.spirio.fr/access.log combined

        Redirect permanent / https://dns.spirio.fr

        # Activer l'emplacement spécifique pour les défis ACME
        <Location "/.well-known/acme-challenge/">
            # Désactiver la gestion d'erreur pour ce chemin
            ErrorDocument 404 "Not Found"

            # Permettre l'accès à ce répertoire
            Require all granted
        </Location>

</VirtualHost>

<VirtualHost *:443>
        ServerName dns.spirio.fr
        ServerAdmin contact@spirio.fr

        ErrorLog /var/log/httpd/002-dns.spirio.fr/error.log
        CustomLog /var/log/httpd/002-dns.spirio.fr/access.log combined

        SSLEngine on

        # Enable proxy modules

        SSLProxyEngine On
        SSLProxyCheckPeerCN Off
        SSLProxyCheckPeerName Off
        SSLProxyVerify none

        ProxyRequests Off
        ProxyPass "/" "https://127.0.0.1:8081/"
        ProxyPassReverse "/" "https://127.0.0.1:8081/"
    
        # Logging
        ErrorLog /var/log/httpd/002-dns.spirio.fr/error.log
        CustomLog /var/log/httpd/002-dns.spirio.fr/access.log combined

        SSLCertificateFile /etc/letsencrypt/live/dns.spirio.fr/fullchain.pem
        SSLCertificateKeyFile /etc/letsencrypt/live/dns.spirio.fr/privkey.pem
</VirtualHost>
```

I generate the certificate with 
```
sudo certbot certonly --preferred-chain "ISRG Root X1" -d dns.spirio.fr
```

**Comment by ThinkChaos** [comment]
Hi,

Please provide more information when you request support, it's virtually impossible to know what's wrong with just configs. The current info isn't enough to even be sure there's an issue with blocky, it could very well just be your firewall.  

So please provide us with your blocky config and logs:
- set `log: level: trace` in blocky's config
- make a request to each port
- post the resulting logs (censored is fine)

**Comment by SpirioSpirio** [comment]
I set log: level: trace in blocky's config. The requests for the port 53 in UDP and TCP are ok.
This is the logs about request on 853
```
kdig -d @54.37.234.80 +tls-ca +tls-host=dns.spirio.fr example3.com (don't works, there is no log for this request)
dig +tls @54.37.234.80 example.com (works)

# My computer
kdig -d @54.37.234.80 +tls-ca +tls-host=dns.spirio.fr example3.com 

;; DEBUG: Querying for owner(example3.com.), class(1), type(1), server(54.37.234.80), port(853), protocol(TCP)
;; DEBUG: TLS, imported 354 system certificates
;; DEBUG: TLS, received certificate hierarchy:
;; WARNING: TLS, handshake failed (The requested data were not available.)
;; ERROR: failed to query server 54.37.234.80@853(TCP)
;; Warning: Message parser reports malformed message packet.

------------------------------------------------------------------------------
dig +tls @54.37.234.80 example.com

;; Warning: Message parser reports malformed message packet.

; <<>> DiG 9.18.20 <<>> +tls @54.37.234.80 example.com
; (1 server found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 4602
;; flags: qr rd ra ad; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 2

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 1232
; PAD: (408 bytes)
;; QUESTION SECTION:
;example.com.			IN	A

;; ANSWER SECTION:
example.com.		1783	IN	A	93.184.216.34

;; ADDITIONAL SECTION:
.			0	RESERVED0 OPT	15 8 AA1DQUNIRUQ=

;; Query time: 104 msec
;; SERVER: 54.37.234.80#853(54.37.234.80) (TLS)
;; WHEN: Wed Dec 13 10:54:17 CET 2023
;; MSG SIZE  rcvd: 491

# podman logs -f blocky
[2023-12-13 10:54:17] DEBUG server: new request
[2023-12-13 10:54:17] TRACE custom_dns_resolver: go to next resolver client_ip=86.202.56.100 client_names=86.202.56.100 question=A (example.com.) resolver=hosts_file
[2023-12-13 10:54:17] DEBUG blacklist_resolver: checking groups for request client_ip=86.202.56.100 client_names=86.202.56.100 groupsToCheck=ads question=A (example.com.)
[2023-12-13 10:54:17] DEBUG caching_resolver: domain 'example.com' was requested 3 times, total cache size: 11 client_ip=86.202.56.100 client_names=86.202.56.100 domain=example.com question=A (example.com.)
[2023-12-13 10:54:17] DEBUG caching_resolver: domain is cached client_ip=86.202.56.100 client_names=86.202.56.100 domain=example.com question=A (example.com.)
[2023-12-13 10:54:17]  INFO queryLog: query resolved answer=A (93.184.216.34) client_ip=86.202.56.100 client_names=86.202.56.100 duration_ms=0 hostname=blocky-hostname question_name=example.com. question_type=A response_code=NOERROR response_reason=CACHED response_type=CACHED
```

**Comment by ThinkChaos** [comment]
Given this partially works, I doubt this is a blocky issue. I'd bet it's something to do with the reverse proxy but I can't really help with that.

Also I ran the `kdig` command and it works for me. Did you fix the issue?  
If yes, please provide info here so anyone else with a similar setup can solve it too.

**Comment by SpirioSpirio** [comment]
Yes, I fixed the issue! I totally changed the network configuration, the pod has now a dedicated IP address and packet from 53 and 853 are sent via NAT configuration. The blocky service is now able to start properly.

The docker-compose configuration offered didn't work for me. I had to set up an IP address and a use of the default podman bridge interface.

This is what I added in case it could help someone else.
```
   networks:
      default:
        ipv4_address: 172.24.0.79

networks:
  default:
    external:
      name: your-network-name
```

Thank you for your help.
